File: java/src/test/java/com/iteso/nintendo/pokemons/LucarioTest.java
package com.iteso.nintendo.pokemons;

import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.assertEquals;

/**
 * Created by rvillalobos on 2/24/18.
 */
public class LucarioTest {
    Lucario lucario;

    @Before
    public void setUp(){
        lucario = new Lucario();
    }

    /** Main attack test. */
    @Test
    public void mainAttackTest() {
        assertEquals("Attacking opponent with Small Fire causing a damage of 3", lucario.mainAttack());
    }

    /** Second attack test. */
    @Test
    public void secondAttackTest() {
        assertEquals("Attacking opponent with Quick Attack causing a damage of 5", lucario.secondAttack());
    }

    /** Defense test */
    @Test
    public void defendTest() {
        assertEquals("Defending attack, damage caused is 0 new HP is 77", lucario.defend(0));
    }

    /** Name test */
    @Test
    public void getNameTest() {
        assertEquals("Lucario", lucario.getName());
    }

    /** Evolve test */
    @Test
    public void evolveTest() {
        assertEquals(null, lucario.evolve());
    }

    /** Type test */
    @Test
    public void typeTest() {
        assertEquals("fighter", lucario.getType());
    }

    /** Evolvution test */
    @Test
    public void hasEvolutionTest() {
        assertEquals(true, lucario.hasEvolution());
    }

    /** Main Damage test */
    @Test
    public void getMainAttackDamageTest() {
        assertEquals(3, lucario.getMainAttackDamage());
    }

    /** Second Damage test */
    @Test
    public void getSecondAttackDamageTest() {
        assertEquals(5, lucario.getSecondAttackDamage());
    }

    /** Hit points test */
    @Test
    public void getHitPointsTest() {
        assertEquals(77, lucario.getHitPoints());
    }

    /** Defense multiplier test */
    @Test
    public void getDefenseMultiplierTest() {
        assertEquals(0.3, lucario.getDefenseMultiplier(), 0.0);
    }
}

File: python/src/unittest/python/LucarioTest.py
import unittest
from Lucario import *


class LucarioTest(unittest.TestCase):
    def setUp(self):
        self.lucario = Lucario()

    def mainAttackTest(self):
        self.assertEquals(
            "Attacking opponent with Small Fire causing a damage of 3",
            self.lucario.mainAttack())

    def secondAttackTest(self):
        self.assertEquals(
            "Attacking opponent with Quick Attack causing a damage of 5",
            self.lucario.secondAttack())

    def defendTest(self):
        self.assertEquals(
            "Defending attack, damage caused is 0 new HP is 77",
            self.lucario.defend(0))

    def getNameTest(self):
        self.assertEquals("Lucario", self.lucario.getName())

    def evolveTest(self):
        self.assertEquals(None, self.lucario.evolve())

    def typeTest(self):
        self.assertEquals("fighter", self.lucario.getType())

    def hasEvolutionTest(self):
        self.assertEquals(True, self.lucario.isHasEvolution())


    def getMainAttackDamageTest(self):
        self.assertEquals(3, self.lucario.getMainAttackDamage())

    def getSecondAttackDamageTest(self):
        self.assertEquals(5, self.lucario.getSecondAttackDamage())

    def getDefenseMultiplierTest(self):
        self.assertEquals(0.3, self.lucario.getDefenseMultiplier())

    def getHitPointsTest(self):
        self.assertEquals(77, self.lucario.getHitPoints())